### PR TITLE
Adds JWT auth flow + updated catalog items + terriajs v8

### PR DESCRIPTION
**Pull request by sverhoeven**
Use latest terriajs for ewatercycle explorer.
Adds JWT auth flow + updated catalog items

Do not merge, this PR is here to show the difference with local copy of main branch of TerriaJS/TerriaMap repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,3 +19,50 @@ This is a complete website built using the TerriaJS library. See the [TerriaJS R
...
+Get terriajs fork with
+
+```shell
+mkdir packages
+git clone git@github.com:eWaterCycle/terriajs.git packages/terriajs
+cd packages/terriajs/
+git checkout ewatercycle-v8
+# edit stuff
+cd ../..
+yarn run gulp
```
Comment (Peter9192) [inline README.md]:
```suggestion
```

Not sure if that works. Just proceed with install below?

File: README.md [hunk 2]
```diff
@@ -19,3 +19,50 @@ This is a complete website built using the TerriaJS library. See the [TerriaJS R
...
+```
+
+See https://docs.terria.io/guide/contributing/development-environment/#building-a-terriamap-against-a-modified-terriajs
+
+## Run
+
+```shell
+yarn install
+yarn run gulp
+yarn start
```
Comment (Peter9192) [inline README.md]: I tried making a local install but keep running into `gyp ERR!`

<details><summary>trace log</summary>

```
...

In file included from /home/peter/.node-gyp/16.3.0/include/node/v8.h:30,
                 from /home/peter/.node-gyp/16.3.0/include/node/node.h:63,
                 from ../../nan/nan.h:58,
                 from ../src/binding.cpp:1:
/home/peter/.node-gyp/16.3.0/include/node/v8-internal.h: In function ‘void v8::internal::PerformCastCheck(T*)’:
/home/peter/.node-gyp/16.3.0/include/node/v8-internal.h:452:38: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  452 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                      ^~~~~~~~~~~
      |                                      remove_cv
/home/peter/.node-gyp/16.3.0/include/node/v8-internal.h:452:38: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  452 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                      ^~~~~~~~~~~
      |                                      remove_cv
/home/peter/.node-gyp/16.3.0/include/node/v8-internal.h:452:50: error: template argument 2 is invalid
  452 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                                  ^
/home/peter/.node-gyp/16.3.0/include/node/v8-internal.h:452:63: error: ‘::Perform’ has not been declared
  452 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                                               ^~~~~~~
../src/binding.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE render(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/binding.cpp:284:98: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
  284 |     int status = uv_queue_work(uv_default_loop(), &ctx_w->request, compile_it, (uv_after_work_cb)MakeCallback);
      |                                                                                                  ^~~~~~~~~~~~
../src/binding.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE render_file(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/binding.cpp:320:98: warning: cast between incompatible function types from ‘void (*)(uv_work_t*)’ {aka ‘void (*)(uv_work_s*)’} to ‘uv_after_work_cb’ {aka ‘void (*)(uv_work_s*, int)’} [-Wcast-function-type]
  320 |     int status = uv_queue_work(uv_default_loop(), &ctx_w->request, compile_it, (uv_after_work_cb)MakeCallback);
      |                                                                                                  ^~~~~~~~~~~~
In file included from ../../nan/nan.h:58,
                 from ../src/binding.cpp:1:
../src/binding.cpp: At global scope:
/home/peter/.node-gyp/16.3.0/include/node/node.h:806:43: warning: cast between incompatible function types from ‘void (*)(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)’ {aka ‘void (*)(v8::Local<v8::Object>)’} to ‘node::addon_register_func’ {aka ‘void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)’} [-Wcast-function-type]
  806 |       (node::addon_register_func) (regfunc),                          \
      |                                           ^
/home/peter/.node-gyp/16.3.0/include/node/node.h:840:3: note: in expansion of macro ‘NODE_MODULE_X’
  840 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
      |   ^~~~~~~~~~~~~
../src/binding.cpp:358:1: note: in expansion of macro ‘NODE_MODULE’
  358 | NODE_MODULE(binding, RegisterModule);
      | ^~~~~~~~~~~
make: *** [binding.target.mk:133: Release/obj.target/binding/src/binding.o] Error 1
make: Leaving directory '/home/peter/ewatercycle/TerriaMap/node_modules/node-sass/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/peter/ewatercycle/TerriaMap/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (node:events:394:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Linux 4.19.128-microsoft-standard
gyp ERR! command "/home/peter/.nvm/versions/node/v16.3.0/bin/node" "/home/peter/ewatercycle/TerriaMap/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd /home/peter/ewatercycle/TerriaMap/node_modules/node-sass
gyp ERR! node -v v16.3.0
```
</details>

File: lib/Views/LauncherAuth.jsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                value={this.state.password}
+                onChange={this.onPasswordChange}
+              ></input>
+            </label>
+            <p>
+              Make sure you logged into{" "}
+              <a href={jupyterHubUrl} target="_blank" className={Styles.link}>
+                Jupyter Hub
+              </a>{" "}
+              at least once with supplied username.
```
Comment (Peter9192) [inline lib/Views/LauncherAuth.jsx:147]: You mean: the launcher doesn't work if you have never (directly) logged into Jupyter before?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ jobs:
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm run gulp lint release
+    - run: yarn install --frozen-lockfile
```
Comment (Peter9192) [inline .github/workflows/ci.yml:22]: What does that mean? I noticed you remove the package lock file. Shouldn't this be tied to a specific version of terriajs?

File: wwwroot/config.json [hunk 5]
```diff
@@ -24,33 +30,33 @@
...
     // See comments in lib/Views/global.scss for further information about customising colors, logos, fonts etc.
     "brandBarElements": [
       "",
-      "<a target=\"_blank\" href=\"https://terria.io\"><img src=\"images/terria_logo.png\" height=\"52\" title=\"Version: {{version}}\" /></a>",
+      "<a target=\"_blank\" href=\"https:/ewatercyclre.org\"><img src=\"https://github.com/eWaterCycle/ewatercycle/raw/main/docs/examples/logo.png\" height=\"52\" title=\"Version: {{version}}\" /></a>",
       ""
     ],
     // Shown when things go wrong.
-    "supportEmail": "help@example.com",
+    "supportEmail": "help@ewatercycle.org",
```
Comment (Peter9192) [inline wwwroot/config.json]: Who owns this eamil address?

File: wwwroot/config.json [hunk 6]
```diff
@@ -24,33 +30,33 @@
...
         },*/
     // This is used in various text messages. You should also set <title> in index.html
-    "appName": "Terria Map",
+    "appName": "eWaterCycle explorer",
 
     // See comments in lib/Views/global.scss for further information about customising colors, logos, fonts etc.
     "brandBarElements": [
       "",
-      "<a target=\"_blank\" href=\"https://terria.io\"><img src=\"images/terria_logo.png\" height=\"52\" title=\"Version: {{version}}\" /></a>",
+      "<a target=\"_blank\" href=\"https:/ewatercyclre.org\"><img src=\"https://github.com/eWaterCycle/ewatercycle/raw/main/docs/examples/logo.png\" height=\"52\" title=\"Version: {{version}}\" /></a>",
```
Comment (Peter9192) [inline wwwroot/config.json]:
```suggestion
      "<a target=\"_blank\" href=\"https:/ewatercycle.org\"><img src=\"https://github.com/eWaterCycle/ewatercycle/raw/main/docs/examples/logo.png\" height=\"52\" title=\"Version: {{version}}\" /></a>",
```
broken link

File: .github/workflows/ci.yml (hunk 4 above)
Comment (Peter9192) [reply .github/workflows/ci.yml:22]: Ah, now I see there is a yarn.lock added instead. I guess that replaces it and is what is used here?

**Review by Peter9192 — COMMENTED** [review]
Tried running locally but failed. Mostly looks good on the hosted explorer though. Wondering why there's two dedicated geojson files in wwwroot/data, but also a lot of geometry in wwwroot/init/ewatercycle.json. Can't we move the geometry to separate files so that the ewatercycle.json becomes more readable?

File: lib/Views/LauncherAuth.jsx (hunk 3 above)
Comment (sverhoeven) [reply lib/Views/LauncherAuth.jsx:147]: yep, the first login prepares JH for access using API.

File: wwwroot/config.json (hunk 5 above)
Comment (sverhoeven) [reply wwwroot/config.json]: Dunno, switched to ewatercycle@esciencecenter.nl which lands a a O365 groups email you are a member of. Email is also used on https://www.ewatercycle.org/contact/

File: README.md (hunk 2 above)
Comment (sverhoeven) [reply README.md]: Terria heeft node v10 nodig. Kan het met deze oude versie proberen?

File: README.md (hunk 1 above)
Comment (sverhoeven) [reply README.md]: do `yarn run gulp` later.

**Comment by sverhoeven** [comment]
I moved the geojson to their own files.

**Review by Peter9192 — APPROVED** [review]
Nice! Thanks for moving the geojson to separate files. I was now able to build and run locally with the updated instructions in the README. 

Only thing I noticed is that when you select a catchment, the preview remains fixed at the BeNeLux. It would be more intuititve if that would automatically adjust, but I guess we could do this in a seperate PR.

**Comment by sverhoeven** [comment]
Happy to hear you can now run it.

I noticed that as well, seems initial miniap is tied to home camera. Created https://github.com/eWaterCycle/TerriaMap/issues/7 for it.

**Comment by sverhoeven** [comment]
Made ewatercycle-v8 the default branch in this repo, so no longer need to merge it

**Pull request closed without merging**